src/python/ensembl/production/datafile/scraper/parsers/__init__.py

```diff
@@ -0,0 +1,39 @@
+#  See the NOTICE file distributed with this work for additional information
+#  regarding copyright ownership.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Package containing file parsers for DataFile Scraper"""
+
+from typing import Optional, Tuple, Type
+
+from .bam_parser import *
+from .embl_parser import *
+from .fasta_parser import *
+from .file_parser import *
+
```

Comment:
I would personally add a `__all__` here or in submodules.

Reply:
I don't see the purpose of defining `__all__` in this case. Could you expand on that?

Reply:
So, is `__all__` is defined in submodule `embl_parser` etc...  we don't retrieve the whole of transitive imports when importing the package. Am I wrong?

Reply:
It depends how you import the package.
`from ensembl.production.datafile.scraper.parsers import *` would import the same names that would have been imported when `parsers` was a single module. And importing `*` unless the module was designed for it, is generally discouraged.

Reply:
And not the transitive ones imported in, say `fast_parser` which include `from typing import Optional`, so you get importing Optional but from another module (that might be wrong statement, I remember in java that could trigger error if import were aliased in subpackages)

Reply:
This is what it happens:
```python
from ensembl.production.datafile.scraper.parsers import *
"Optional" in globals()   # This statement is True
```

The result would have been exactly the same when `parsers` was a module, but we're not defining `__all__` for each module.

Reply:
Ok all fine then.